### PR TITLE
Check Java formatting in format.sh

### PR DESCRIPTION
**Pull request by mklim**
Uses google-java-format, a tool that's already in place in
flutter/plugins. This should take care of minor style guide issues and
unresolved imports.

### CODE REVIEW

**Comment by mklim** [comment]
I can see this running in CI here: https://api.cirrus-ci.com/v1/task/4871629293486080/logs/format.log. It's not complaining because it's only running the tool against files that have been changed, and this patch doesn't touch any Java files.

I've applied this locally on my machine against all java files in the repo at flutter/engine#16501.

File: DEPS [hunk 1]
```diff
@@ -436,6 +436,20 @@ deps = {
...
+        'package': 'flutter/android/google-java-format',
+        'version': 'version:1.7'
+       }
+     ],
+     # We want to be able to format these as part of CI, and the CI step that
+     # checks formatting runs without downloading the rest of the Android build
+     # tooling. Therefore unlike all the other Android-related tools, we want to
+     # download this every time.
+     'dep_type': 'cipd',
+   },
```
Comment (dnfield) [inline DEPS]: Can we add a script somewhere to make it easier to create the CIPD package for this?  Or at least instructions?

Comment (mklim) [reply DEPS]: Yes, happy to. Do you have any suggestions, should it be a Wiki page? I have a local `cipd.yaml` on my machine, but I'm not sure where or how to store that information in the repo.

Comment (dnfield) [reply DEPS]: I would either make a README.md for it, add a comment to this part of the deps, or add a script in `tools/` that automates it. You probably don't need a cipd.yaml, you could just have the whole cipd invocatoin in a shell script.

Comment (mklim) [reply DEPS]: Actually I have an idea. I'll put the `cipd.yaml` and a `README` in the CIPD package itself, so anybody who goes into the directory to look into it sees what they need to create a new version.

**Comment by cbracken** [comment]
Fixes the unused imports bit of https://github.com/flutter/flutter/issues/50353

File: DEPS (hunk 1 above)
Comment (mklim) [reply DEPS]: Done.

**Pull request merged**